ucp/_libs/ucx_api.pyx

```diff
@@ -373,6 +380,67 @@ def _ucx_worker_handle_finalizer(
 
 
 IF CY_UCP_AM_SUPPORTED:
+    cdef void _am_recv_completed_callback(
+        void *request,
+        ucs_status_t status,
+        size_t length,
+        void *user_data
+    ):
+        cdef bytearray buf
+        cdef UCXRequest req
+        cdef dict req_info
+        cdef str name, ucx_status_msg, msg
+        cdef set inflight_msgs
+        cdef tuple cb_args
+        cdef dict cb_kwargs
+
+        logger.debug(
+            "_am_recv_completed_callback status %d len %d buf %s" % (
+                status, length, hex(int(<uintptr_t>user_data))
+            )
+        )
+
+        assert user_data != NULL
+
+        with log_errors():
+            req = UCXRequest(<uintptr_t><void*> request)
+            assert not req.closed()
+            req_info = <dict>req._handle.info
+            req_info["status"] = "finished"
+
+            if "cb_func" not in req_info:
+                # This callback function was called before _am_recv_callback() returned
```

Comment:
```suggestion
                # This callback function was called before _am_recv_completed_callback() returned
```

Reply:
Done.

Reply:
I actually realized now that it was correct before. The function here is `_am_recv_completed_callback`, and it needs to be called after `_am_recv_callback` returns.